packages/core/src/__tests__/ceramic-api.test.ts

```diff
@@ -38,12 +38,14 @@ const createIPFS =(overrideConfig: Record<string, unknown> = {}): Promise<IPFSAp
  * document is anchored
  * @param doc
  */
-const syncDoc = async (doc: any): Promise<void> => {
-  await new Promise(resolve => {
+const anchorDoc = async (ceramic: Ceramic, doc: any): Promise<void> => {
+  const p = new Promise(resolve => {
```

Comment:
I did this a little differently than how it works in ceramic-anchor-test, but I think this way is more correct?  If the concern is that the doc could get anchored before we set up the change listener, then I think the way ceramic-anchor-test does it (`anchor()` then immediately `syncDoc()`) makes the race window smaller, but doesn't completely eliminate it, whereas this way does by ensuring we're already listening for a change before requesting an anchor.  Does that make sense to you @simonovic86?  Should I update ceramic-anchor-test to do the same?